src/components/studies/StudyBuilder.tsx

```diff
@@ -391,24 +390,19 @@ const StudyBuilder: FunctionComponent<StudyBuilderProps> = ({
                         hasObjectChanged={hasObjectChanged}
                         saveLoader={saveLoader}
                         id={id}
-                        currentAppDesign={
-                          builderInfo.study.clientData?.appDesign ||
-                          ({} as StudyAppDesign)
-                        }
-                        onSave={() => {
+                        study={builderInfo.study}
+                        onSave={(updatedStudy: Study) => {
+                          setData({
```

Comment:
saveStudy does setData. Do we need to call setData again?

Reply:
Updated.